src/channel_state.ts

```diff
@@ -509,14 +468,16 @@ export class ChannelState<
         if (!idMatch) {
           continue;
         }
-        let messageWithReaction = this._removeReactionFromMessage(msg, reaction);
-        messageWithReaction = messageWithReaction.updateIn(
-          ['reaction_counts', reaction.type],
-          (old: number) => (old ? old - 1 : 0),
-        );
+        const messageWithReaction = this._removeReactionFromMessage(msg, reaction);
+        const oldReactionCounts = messageWithReaction.reaction_counts;
+        if (oldReactionCounts) {
+          const oldReactionType = oldReactionCounts[reaction.type];
+          oldReactionCounts[reaction.type] = oldReactionType ? oldReactionType - 1 : 0;
+          messageWithReaction.reaction_counts = oldReactionCounts;
```

Comment:
Looks like this is a responsibility of `_removeReactionFromMessage` which would also prevent duplication below

Pointing this but it might actually be out of scope of this PR.

Reply:
hmm yeah, I think maybe it's for another PR so we can keep the scope of this to immutables

Reply:
Good idea, SGTM